tools/tuhi-kete.py

```diff
@@ -601,6 +611,9 @@ def do_exit(self, args):
         return True
 
     def precmd(self, line):
+        if self.want_to_stop.is_set():
+            return 'exit'
+
```

Comment:
I am pretty sure we can do better and exit the shell if we receive the event, without waiting for user input.
I'll try to figure out how.